wrappers/python/src/swig_doxygen/swig_lib/python/extend.i

```diff
@@ -55,26 +55,28 @@
          - getEnergy (bool=False) whether to store potential and kinetic energy in the State
          - getParameter (bool=False) whether to store context parameters in the State
          - enforcePeriodicBox (bool=False) if false, the position of each particle will be whatever position is stored in the Context, regardless of periodic boundary conditions.  If true, particle positions will be translated so the center of every molecule lies in the same periodic box.
-         - groups (int=-1) a set of bit flags for which force groups to include when computing forces and energies.  Group i will be included if (groups&(1<<i)) != 0.  The default value includes all groups.
+         - groups (set={0,1,2,...,31}) a set of indices for which force groups
+           to include when computing forces and energies. The default value
+           includes all groups. groups can also be passed as an unsigned integer
+           interpreted as a bitmask, in which case group i will be included if
+           (groups&(1<<i)) != 0.
         """
-        
-        if getPositions: getP=1
-        else: getP=0
-        if getVelocities: getV=1
-        else: getV=0
-        if getForces: getF=1
-        else: getF=0
-        if getEnergy: getE=1
-        else: getE=0
-        if getParameters: getPa=1
-        else: getPa=0
-        if enforcePeriodicBox: enforcePeriodic=1
-        else: enforcePeriodic=0
+        getP, getV, getF, getE, getPa, enforcePeriodic = map(bool,
+            (getPositions, getVelocities, getForces, getEnergy, getParameters,
+             enforcePeriodicBox))
+
+         try:
+             # is the input integer-like?
+             groups_mask = int(groups)
+         except TypeError:
+             # nope, okay, then it should be an iterable
+             groups_mask = functools.reduce(operator.or_,
+                     ((1<<x) & 0xffffffff for x in set(groups)))
```

Comment:
> I mean that adding support for -1-handling in the set doesn't add any new functionality.

Got it.

> Also, it seems contrary to the purpose of passing in a set from Python, which is to avoid dealing with bit-twiddling and the binary representation of integers. To understand why -1 means "all forcegroups", you need to know how two's complement works.

Agreed.  Hence why I said I have no problems with it :).  I was just remarking because it was a change in functionality from one commit to the next.